mtags/src/main/scala/scala/meta/internal/semver/SemVer.scala

```diff
@@ -1,27 +1,71 @@
 package scala.meta.internal.semver
 
+import scala.util.Try
+
 object SemVer {
 
-  case class Version(major: Int, minor: Int, patch: Int) {
+  case class Version(
+      major: Int,
+      minor: Int,
+      patch: Int,
+      releaseCandidate: Option[Int],
+      milestone: Option[Int]
+  ) {
     def >(that: Version): Boolean = {
       this.major > that.major ||
       (this.major == that.major && this.minor > that.minor) ||
-      (this.major == that.major && this.minor == that.minor && this.patch > that.patch)
+      (this.major == that.major && this.minor == that.minor && this.patch > that.patch) ||
+      // 3.0.0 > 3.0.0-M2 and 3.0.0 > 3.0.0-RC1
+      (this.milestone.isEmpty && this.releaseCandidate.isEmpty && (that.milestone.isDefined || that.releaseCandidate.isDefined)) ||
+      // 3.0.0-RC2 > 3.0.0-RC1
+      comparePreRelease(that, (v: Version) => v.releaseCandidate) ||
+      // 3.0.0-M2 > 3.0.0-M1
+      comparePreRelease(that, (v: Version) => v.milestone)
```

Comment:
Sorry, I think I gave a wrong hint...
How about comparing versions with different pre-release versions? The following test should pass, but I guess it fails now.
 
```
    assert(
      SemVer.isLaterScalaVersion("3.0.0-M1", "3.0.0-RC1")
    )
```

We have to compare `releaseCandidate` and `milestone` simultaneously.
Or store the label (like `RC` or `M` part) to `Version` and compare those labels by ASCII sort order (this would be more generic implementation).

```
  case class Version(
      major: Int,
      minor: Int,
      patch: Int,
      preRelease: Option[(String, Int)], // something like this?
  ) 
```

> Identifiers with letters or hyphens are compared lexically in ASCII sort order.
> https://semver.org/#spec-item-11

Reply:
ach, righ!. Every time I think I got it right, there are still edge cases. I will add tests for that too and try figure it out. Good catch.

Reply:
Added additional check:
```scala
      // 3.0.0-RC1 > 3.0.0-M1
      this.releaseCandidate.isDefined && that.milestone.isDefined ||
```

Reply:
lol so many checks. Great job on figuring these out.

Reply:
Yeah, that's why I was not keen on options at first. But I think with the comments it's not that bad. And I added a lot of tests :sweat_smile: